app/controllers/admin/config/branding_controller.rb

```diff
@@ -38,4 +38,36 @@ def logo
       end
     end
   end
+
+  def fonts
+    previous_default_text_size = SiteSetting.default_text_size
+    SiteSetting::Update.call(
+      guardian:,
+      params: {
+        settings: [
+          { setting_name: "base_font", value: params[:base_font] },
+          { setting_name: "heading_font", value: params[:heading_font] },
+          {
+            setting_name: "default_text_size",
+            value: params[:default_text_size],
+            backfill: params[:update_existing_users].present?,
+          },
+        ],
+      },
+    ) do
+      on_success { render json: success_json }
+      on_failed_policy(:settings_are_visible) do |policy|
+        raise Discourse::InvalidParameters, policy.reason
+      end
+      on_failed_policy(:settings_are_unshadowed_globally) do |policy|
+        raise Discourse::InvalidParameters, policy.reason
+      end
+      on_failed_policy(:settings_are_configurable) do |policy|
+        raise Discourse::InvalidParameters, policy.reason
+      end
+      on_failed_policy(:values_are_valid) do |policy|
+        raise Discourse::InvalidParameters, policy.reason
+      end
```

Comment:
Not really related to this change but there is alot of repetition here. Would be nice to be able to do the following instead.

```suggestion
      on_failed_policy(:settings_are_unshadowed_globally, :settings_are_visible, :settings_are_configurable, :values_are_valid) do |policy|
        raise Discourse::InvalidParameters, policy.reason
      end
```